### PR TITLE
Apply introductory offer on on the flow that starts from move page

### PR DESCRIPTION
**Pull request by Imran92**
Related to https://github.com/Automattic/dotcom-forge/issues/8280

## Proposed Changes

* If the user comes to the checkout page with the flow beginning from the `move-lp` (https://wordpress.com/move) page, we mark the site with the introductory sticker to apply the introductory offer discount on the Creator plan.

I haven't written tests for it yet to get feedback first and unblock (check the thread link on the issue) other tasks.

image

## Why are these changes being made?

* To make our introductory offer on migration more visible and actually apply the discount.

## Testing Instructions

- Apply the diff D155293-code
- Make sure you are sandboxed
- Go to https://wordpress.com/move
- Make sure you see the discount on the Creator plan
- Click on the "Get Creator" button
- Complete the checkout flow and make sure you saw the discount applied

Try the steps above as both logged-in and logged-out users. If "https://wordpress.com/start/business/?ref=move-lp" shows error, replace https://wordpress.com with calypso.localhost:3000

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/checkout/src/components/checkout-main.tsx [hunk 1]
```diff
@@ -23,6 +23,7 @@ import { errorNotice, infoNotice } from 'calypso/state/notices/actions';
 import isPrivateSite from 'calypso/state/selectors/is-private-site';
 import isAtomicSite from 'calypso/state/selectors/is-site-automated-transfer';
 import { isJetpackSite } from 'calypso/state/sites/selectors';
+import { useCheckoutMigrationIntroductoryOfferSticker } from '../../../../data/site-migration/use-checkout-migration-introductory-offer-sticker';
```
Comment (m1r0) [inline client/my-sites/checkout/src/components/checkout-main.tsx]:
```suggestion
import { useCheckoutMigrationIntroductoryOfferSticker } from 'calypso/data/site-migration/use-checkout-migration-introductory-offer-sticker';
```

This is a bit easier to read. WDYT?

**Review by m1r0 — APPROVED** [review]
Nice! I'm assuming this took some digging, so kudos for the implementation! 🙌 I have just one small suggestion but the overall approach looks good to me so I'm approving to unblock the email campaign. We can iterate/add tests in future PRs.

Also, here are some questions that came to mind:

- Should the discount be applied to the two-year plan as well? Similar to what Dimitry did in D155114-code.
- I was expecting this to lead me to the migration flow. Are we skipping the migration flow?

**Comment by Imran92** [comment]
Thanks a lot for the review @m1r0 !! <3

> Should the discount be applied to the two-year plan as well? Similar to what Dimitry did in [D155114-code](https://href.li/?https://code.a8c.com/D155114).

My quick assumption is it'll automatically get applied. But if it doesn't, we can add a small iteration.

> I was expecting this to lead me to the migration flow. Are we skipping the migration flow?

I didn't change anything about the flow here, so it's leading to the same flow as it used to do I think 🤔 The only change this PR should make is applying the discount in the existing flow that begins from the Creator product button on the move page.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16331716

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @Imran92 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16331716).

**Pull request merged**